### ISSUE TITLE
@MockBean not applying AOP advice @Around for classes with kotlin coroutines/suspend functions

### ISSUE THREAD

**Issue by frehov** [micronaut-projects/micronaut-test#610]
### Expected Behavior

When writing a service with suspended methods (coroutines), I'd expect the service to be created and be able to mock the service implementation with `@MockBean`.

### Actual Behaviour

When writing a service with suspended methods (coroutines), The service fails to compile in test if using `@MockBean` on the service class containing suspended functions

If you add `open` to the class being mocked, the service will appear to be mocked, but rather it will be created from the original bean definition. Adding `open` to both class and method ensures that the mock is used.

### Steps To Reproduce

1. Run gradle test on the example application.
2. Compilation fails.
4. See explanation and workarounds in example application for more information

### Environment Information

- Windows 11, 22H1
- JDK 17, eclipse temurin/adoptium

### Example Application

https://github.com/frehov/micronaut-kotlin-suspend-mockbean-failure

### Version

3.5.2

**Comment by ojebari-ma** [comment]
Can you please confirm that this is still an issue in the latest version of Micronaut?

**Comment by frehov** [comment]
@ojebari-ma yes, still an issue as far as I can tell, and testing with steps described in readme resolves the below error.
I've brought it forward, but am having some trouble getting the test to validate the mock.
I've not updated the example, but can do so with the test being broken.

```log
[INFO] Processing java sources with annotation processors: C:\Users\Fredrik Hov\workspace\micronaut\micronaut-kotlin-suspend-mockbean-failure\build\tmp\kapt3\stubs\test\com\example\micronaut\controllers\StatusControllerTest.java, C:\Users\Fredrik Hov\workspace\micronaut\micronaut-kotlin-suspend-mockbean-failure\build\tmp\kapt3\stubs\test\error\NonExistentClass.java
[INFO] Processing types with annotation processors: 
C:\Users\Fredrik Hov\workspace\micronaut\micronaut-kotlin-suspend-mockbean-failure\build\tmp\kapt3\stubs\test\com\example\micronaut\controllers\StatusControllerTest.java:46: error: Cannot apply AOP advice to final class. Class must be made non-final to support proxying: com.example.micronaut.services.StatusService
    public final com.example.micronaut.services.StatusService connectionStatusService() {
                                                              ^
```
